src/suites/cts/validation/render_pass_descriptor.spec.ts

```diff
@@ -0,0 +1,554 @@
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  await t.tryRenderPass(false, descriptor);
+});
+
+g.test('it is invalid to use a resolve target whose usage is not output attachment', async t => {
+  const multisampledColorTexture = t.createTexture({ sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({
+    usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
+  });
+
+  const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+  colorAttachment.resolveTarget = resolveTargetTexture.createView();
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  await t.tryRenderPass(false, descriptor);
+});
+
+g.test('it is invalid to use a resolve target in error state', async t => {
+  const ARRAY_LAYER_COUNT = 1;
+
+  const multisampledColorTexture = t.createTexture({ sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({ arrayLayerCount: ARRAY_LAYER_COUNT });
+
+  const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+  await t.expectValidationError(() => {
+    colorAttachment.resolveTarget = resolveTargetTexture.createView({
+      dimension: '2d',
+      format: 'rgba8unorm',
+      baseArrayLayer: ARRAY_LAYER_COUNT + 1,
+    });
+  });
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  await t.tryRenderPass(false, descriptor);
+});
+
+g.test('use of multisampled attachment and non multisampled resolve target is allowed', async t => {
+  const multisampledColorTexture = t.createTexture({ sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({ sampleCount: 1 });
+
+  const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+  colorAttachment.resolveTarget = resolveTargetTexture.createView();
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  t.tryRenderPass(true, descriptor);
+});
+
+g.test('use a resolve target in a format different than the attachment is not allowed', async t => {
+  const multisampledColorTexture = t.createTexture({ sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({ format: 'bgra8unorm' });
+
+  const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+  colorAttachment.resolveTarget = resolveTargetTexture.createView();
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  await t.tryRenderPass(false, descriptor);
+});
+
+g.test('size of the resolve target must be the same as the color attachment', async t => {
+  const size = 16;
+  const multisampledColorTexture = t.createTexture({ width: size, height: size, sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({
+    width: size * 2,
+    height: size * 2,
+    mipLevelCount: 2,
+  });
+
+  {
+    const resolveTargetTextureView = resolveTargetTexture.createView({ baseMipLevel: 0 });
```

Comment:
I think you want to also specify `mipLevelCount: 1` here to avoid triggering two different error checks.

Reply:
Indeed!